crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -75,13 +75,19 @@ def current_step_state
       progress[:steps][current_step] || {}
     end
 
-    def start_step
+    # 'step' is name of the step user wants to start.
+    # If `out_of_order` is true, the step could be executed out of order
+    def start_step(step_name, out_of_order = false)
       ::Crowbar::Lock::LocalBlocking.with_lock(shared: false, logger: @logger, path: lock_path) do
-        if running?
+        if running? step_name
           @logger.warn("The step has already been started")
```

Comment:
This needs changing to `"The #{current_step} step has already been started"`, because with this API change the already-running step can now be different to the one the caller is requesting to start.